src/realm/object-store/c_api/realm.cpp

```diff
@@ -121,6 +121,20 @@ RLM_API bool realm_rollback(realm_t* realm)
     });
 }
 
+RLM_API uint64_t realm_add_realm_changed_callback(realm_t* realm, realm_on_realm_change_func_t callback,
+                                                  void* userdata, realm_free_userdata_func_t free_userdata)
+{
+    util::UniqueFunction<void()> func = [callback, userdata = UserdataPtr{userdata, free_userdata}]() {
+        callback(userdata.get());
+    };
+    return CBindingContext::get(*realm).realm_changed_callbacks().add(std::move(func));
```

Comment:
I haven't tested this, but if you're using the binding context in realm-java then this should give you the same experience in realm-kotlin because it's based on the same binding context infrastructure.

Reply:
Yes, we also use `did_change` in Java. It should be guaranteed to fire before any other notifications are triggered.

Reply:
Could this be added in the C-API docs, so potential reimplementations would catch this requirement?